tests/integration/records/single_db/test_records_unload_datetime.py

```diff
@@ -0,0 +1,370 @@
+                if 'pandas' in str(e) and expect_pandas_failure:
+                    # as expected
+                    continue
+                else:
+                    raise
+            self.assertFalse(expect_pandas_failure)
+
+    def test_unload_datetimetz(self) -> None:
+        self.datetime_fixture.createDateTimeTzTable()
+        matching_dateformat = {
+            'YYYY-MM-DD HH:MI:SS': 'YYYY-MM-DD',
+            'YYYY-MM-DD HH12:MI AM': 'YYYY-MM-DD',
+            'MM/DD/YY HH24:MI': 'MM/DD/YY',
+            'YYYY-MM-DD HH24:MI:SSOF': 'YYYY-MM-DD',
+            'YYYY-MM-DD HH:MI:SSOF': 'YYYY-MM-DD',
+        }
+        for datetimeformattz in DATETIMETZ_CASES:
+            addl_hints: PartialRecordsHints = {}
+            pandas_compatible_addl_hints: PartialRecordsHints = {
+                'dateformat': matching_dateformat[datetimeformattz],
+                'datetimeformat': datetimeformattz.replace('OF', ''),
+            }
+            uses_pandas = False
+            if self.engine.name == 'redshift':
+                if datetimeformattz != 'YYYY-MM-DD HH:MI:SSOF':
+                    # this is the only format supported by Redshift on
+                    # export
+                    uses_pandas = True
+            elif self.engine.name == 'mysql':
+                # mysql has no exporter defined, so everything is via Pandas
+                uses_pandas = True
+            elif self.engine.name == 'postgresql':
+                if datetimeformattz in ['YYYY-MM-DD HH:MI:SSOF',
+                                        'YYYY-MM-DD HH24:MI:SSOF']:
+                    # ensure we keep other areas of ISO style
+                    addl_hints = {
+                        'timeonlyformat': 'HH:MI:SS',
+                        'dateformat': 'YYYY-MM-DD',
+                        'datetimeformat': 'YYYY-MM-DD HH:MI:SS',
+                    }
+                else:
+                    uses_pandas = True
+            elif self.engine.name == 'vertica':
+                # Make sure our '\n' strings below are valid when comparing output
+                addl_hints['record-terminator'] = '\n'
+                if datetimeformattz not in ['YYYY-MM-DD HH:MI:SSOF',
+                                            'YYYY-MM-DD HH24:MI:SSOF']:
+                    uses_pandas = True
+            elif self.engine.name == 'bigquery':
+                # All current export is via Avro
+                uses_pandas = True
+            if uses_pandas:
+                # We're going to need to be sure to use hints that
+                # work in Pandas
+                addl_hints.update(pandas_compatible_addl_hints)
+                if 'AM' in datetimeformattz:
+                    # Pandas is more limited in hint support than it
+                    # should be:
+                    #
+                    # https://github.com/bluelabsio/records-mover/issues/143
+                    # TODO: Should expect failure and complain if it doesn't fail
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker